tests/register/register.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package register
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/coreos/init/tests/util"
+)
+
+type Test struct {
+	Name            string
+	Func            func(*testing.T, Test)
+	BinaryPath      string
+	IgnitionConfig  *string
+	CloudinitConfig *string
+}
+
+func (test Test) Run(t *testing.T) {
+	if os.Getenv("TMPDIR") == "" {
+		tmpDir, err := ioutil.TempDir("/var/tmp", "")
+		if err != nil {
+			t.Fatalf("failed to create temp working dir in /var/tmp: %v", err)
+		}
+
+		err = os.Setenv("TMPDIR", tmpDir)
+		if err != nil {
+			t.Fatalf("couldn't set TMPDIR env var: %v", err)
+		}
+
+		defer test.RemoveAll(t, tmpDir)
```

Comment:
Idiomatically, should be immediately after the error check on the `TempDir` call.